player/lua/select.lua

```diff
@@ -301,11 +270,14 @@ mp.add_key_binding(nil, "select-subtitle-line", function ()
     local duration = mp.get_property_native("duration", math.huge)
 
     -- Strip HTML and ASS tags.
-    for line in r.stdout:gsub("<.->", ""):gsub("{\\.-}", ""):gmatch("[^\n]+") do
+    for line in data:gsub("<.->", ""):gsub("{\\.-}", ""):gmatch("[^\n]+") do
         -- ffmpeg outputs LRCs with minutes > 60 instead of adding hours.
-        sub_times[#sub_times + 1] = line:match("%d+") * 60 + line:match(":([%d%.]*)")
-        sub_lines[#sub_lines + 1] = format_time(sub_times[#sub_times], duration) ..
-                                    " " .. line:gsub(".*]", "", 1)
+        local sub_line = line:gsub(".*]", "", 1)
+        if sub_line ~= "" and sub_line:match("^%s+$") == nil then
```

Comment:
Personally I would rather keep showing empty lines because they are used to separate verses and choruses in LRC. I'm indifferent about lines containing only whitespace.

Reply:
Please split any `async` changes from display changes or other changes not related to async conversion to another commits. (same pr is fine, but we have to have visibility what is changing)

Reply:
Can you keep them `if sub.external and sub["external-filename"]:lower():find("%.lrc$")` at least? It's not perfect but unfortunately the codec returned by ffmpeg seems to be the same as srt.

Reply:
> Can you keep them if sub.external and sub["external-filename"]:lower():find("%.lrc$") at least? It's not perfect but unfortunately the codec returned by ffmpeg seems to be the same as srt.

Yeah, there's no good way to distinguish the lrc format. I noticed it's recognized as subrip by ffmpeg, so maybe it's possible to preserve blank lines against subrip. I don't think regular srt format subtitles usually have blank lines in them!

Reply:
> > Can you keep them if sub.external and sub["external-filename"]:lower():find("%.lrc$") at least? It's not perfect but unfortunately the codec returned by ffmpeg seems to be the same as srt.
> 
> Yeah, there's no good way to distinguish the lrc format. I noticed it's recognized as subrip by ffmpeg, so maybe it's possible to preserve blank lines against subrip. I don't think regular srt format subtitles usually have blank lines in them.

Implemented the idea, let me know what you think.

Reply:
It works great, thanks.